src/Helper/MyParcelCollection.php

```diff
@@ -172,11 +172,26 @@ public function addConsignment(AbstractConsignment $consignment): self
         if ($consignment->getApiKey() === null) {
             throw new MissingFieldException('First set the API key with setApiKey() before running addConsignment()');
         }
-
         $consignment->validate();
-
         $this->push($consignment);
 
+        // Check if all consignments have the same carrier
+        $carrierIds = array_map(function($c) { return $c->getCarrierId(); }, $this->items);
+        if (count(array_unique($carrierIds)) > 1) {
+            throw new \Exception('All consignments in a multi collo shipment must have the same carrier.');
+        }
+
+        // Only set multi collo if all reference_identifiers are the same (or not set)
```

Comment:
I understand the urge to ‘automate’ it here, but
- now the addMultiCollo method has gone, this is a breaking change that we would like to prevent
- a consuming application should (imo) have the choice to add consignments as multi collo or not (like the current situation)
- the check whether the carrier has multi collo ability is missing I think

Reply:
My understanding from the talk was that
- you refactor the addDataList (or something like that) to a method that accepts a collection itself and can set this collection as a multi collo consignment as such (with different weights and options)
- keep the addMultiCollo(consignment, amount) and realy it to your new method (with weight equally distributed) for backwards compatibility